controller/scan/registry.go

```diff
@@ -879,51 +879,114 @@ func (rs *Registry) checkAndPutImageResult(sctx *scanContext, id string, result
 		// not canceled, continue
 	}
 
+	if result.ScanTypesRequested == nil {
+		result.ScanTypesRequested = &share.ScanTypeMap{}
+	}
+
 	if sum, ok := rs.summary[id]; ok {
 		sum.ScannedAt = time.Now().UTC()
-
 		sum.Provider = result.Provider
 		sum.BaseOS = result.Namespace
 		sum.Version = result.Version
-		sum.Result = result.Error
-		if result.Error == share.ScanErrorCode_ScanErrNone {
-			sum.Status = api.ScanStatusFinished
-		} else if result.Error == share.ScanErrorCode_ScanErrNotSupport {
-			sum.Status = api.ScanStatusFinished
-		} else if retAction == scheduler.TaskActionRequeue {
-			sum.Status = api.ScanStatusScheduled
-		} else if result.Error == share.ScanErrorCode_ScanErrSignatureScanError {
-			sum.Status = api.ScanStatusFailed_SignatureScan
-		} else {
-			sum.Status = api.ScanStatusFailed
-		}
 		sum.Author = result.Author
-		sum.Size = result.Size
+		if result.ScanTypesRequested.Vulnerability {
+			sum.Size = result.Size
+		}
 
-		if sum.Status == api.ScanStatusFinished {
-			sum.ScanFlags |= share.ScanFlagCVE
-			if len(result.Layers) != 0 {
-				sum.ScanFlags |= share.ScanFlagLayers
+		// handle status for vuln scan
+		if result.ScanTypesRequested.Vulnerability {
```

Comment:
in https://github.com/neuvector/scanner/pull/85, when is result.Error set to share.ScanErrorCode_ScanErrSignatureScanError ?
(line 905 in this file checks about share.ScanErrorCode_ScanErrSignatureScanError)